python/ray/data/preprocessor.py

```diff
@@ -280,3 +298,8 @@ def _transform_pandas(self, df: "pd.DataFrame") -> "pd.DataFrame":
     def _transform_arrow(self, table: "pyarrow.Table") -> "pyarrow.Table":
         """Run the transformation on a data batch in a PyArrow Table format."""
         raise NotImplementedError()
+
+    @DeveloperAPI
+    def _transform_numpy(self, np_array: "np.ndarray") -> "np.ndarray":
```

Comment:
I _think_ that we would want to support the full NumPy narrow waist here, including `Dict[str, np.ndarray]` for multi-column tables.
```suggestion
    def _transform_numpy(self, np_array: Union["np.ndarray", Dict[str, "np.ndarray"]]) -> Union["np.ndarray", Dict[str, "np.ndarray"]]:
```

Reply:
yep you're totally right on this, i wrote this signature first and only realized we need both later. I will updated it in newer commits